csharp/src/AddressBook/Addressbook.cs

```diff
@@ -77,9 +77,18 @@ public sealed partial class Person : pb::IMessage<Person>, global::System.IEquat
     }
 
     public Person() { }
+
     public Person(Person other) {
```

Comment:
Two reasons: 1) Yes, I think the aliasing could be confusing. 2) As an implementation detail, it means we can keep the repeated field fields (if you see what I mean) readonly, which we couldn't easily do with Clone otherwise. (There are ways, but they'd be ugly.)

Overall, I think MergeFrom(Message) is unlikely to be very useful, and I'd be happy to consider alternatives that didn't require it...